pkg/neorpc/rpcevent/filter.go

```diff
@@ -66,7 +68,29 @@ func Matches(f Comparator, r Container) bool {
 		notification := r.EventPayload().(*state.ContainedNotificationEvent)
 		hashOk := filt.Contract == nil || notification.ScriptHash.Equals(*filt.Contract)
 		nameOk := filt.Name == nil || notification.Name == *filt.Name
-		return hashOk && nameOk
+		parametersOk := true
+		if len(filt.Parameters) > 0 {
```

Comment:
Use nil comparison instead of len comparison because filtering for empty parameter list must also be supported.

Reply:
Not relevant with https://github.com/nspcc-dev/neo-go/pull/3689#discussion_r1851956004.